include/drivers/clock_control.h

```diff
@@ -231,6 +236,43 @@ static inline int clock_control_get_rate(const struct device *dev,
 	return api->get_rate(dev, sys, rate);
 }
 
+/**
+ * @brief Set the rate of the clock controlled by the device.
+ *
+ * On success, the new clock rate is set and ready when this function
+ * returns. This function may sleep, and thus can only be called from
+ * thread context.
+ *
+ * @param dev Device structure whose driver controls the clock.
+ * @param sys Opaque data representing the clock.
+ * @param rate Subsystem clock rate.
```

Comment:
is this in Hz or opaque, vendor specific value?

Reply:
I believe it is easier to use if it is vendor specific.

Reply:
opinions ?

Reply:
I also think that this could be vendor specific, So maybe rephrase to `Opaque data representing clock rate`

Reply:
Vendor specific makes sense to me as well

Reply:
cool ! Thanks for the feedback @teburd.

@nordic-krch can you take a look on it again ?